src/OpenTelemetry/Trace/Configuration/OpenTelemetrySdk.cs

```diff
@@ -106,5 +100,35 @@ public static void EnableOpenTelemetry(Action<OpenTelemetryBuilder> configureOpe
 
             ActivitySource.AddActivityListener(listener);
         }
+
+        internal static void BuildSamplingParameters(
+            in ActivityCreationOptions<ActivityContext> options, out ActivitySamplingParameters samplingParameters)
+        {
+            ActivityContext parentContext = options.Parent;
+            if (parentContext == default)
+            {
+                // Check if there is already a parent for the current activity.
+                var parentActivity = Activity.Current;
+                if (parentActivity != null)
+                {
+                    parentContext = parentActivity.Context;
+                }
+            }
```

Comment:
PR https://github.com/dotnet/runtime/pull/37185 removes the need for this part of the code.

Reply:
I am not sure if you'll merge this OT PR as is till getting the .NET change. but if you do, please be aware that `parentActivity.Context` is possible to throw if the parent was created with defaulted trace id. my fix in .NET is taking care of this issue too to guarantee `Activity.Context` not throw under any condition.

Reply:
Thanks for the heads up @tarekgh - I think that for now it is better to go as it is so we can keep experimenting with the `ActivitySource` this code will be removed as soon as we have access to https://github.com/dotnet/runtime/pull/37185